pkg/controller/elasticquota.go

```diff
@@ -0,0 +1,348 @@
+	for i := 0; i < workers; i++ {
+		go wait.Until(ctrl.run, time.Second, stopCh)
+	}
+	<-stopCh
+	klog.V(2).Info("Shutting down elastic quota workers")
+}
+
+// WithFakeRecorder will set a fake recorder.It is usually used for unit testing
+func WithFakeRecorder(bufferSize int) func(ctrl *ElasticQuotaController) {
+	return func(ctrl *ElasticQuotaController) {
+		ctrl.recorder = record.NewFakeRecorder(bufferSize)
+	}
+}
+
+func (ctrl *ElasticQuotaController) run() {
+	for ctrl.sync() {
+	}
+}
+
+// sync deals with one key off the queue.  It returns false when it's time to quit.
+func (ctrl *ElasticQuotaController) sync() bool {
+	keyObj, quit := ctrl.eqQueue.Get()
+	if quit {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer ctrl.eqQueue.Done(obj)
+		key, ok := obj.(string)
+		if !ok {
+			ctrl.eqQueue.Forget(obj)
+			runtime.HandleError(fmt.Errorf("expected string in workqueue but got %#v", keyObj))
+			return nil
+		}
+		if err := ctrl.syncHandler(key); err != nil {
+			return fmt.Errorf("error syncing elastic quota %s: %s", key, err.Error())
+		}
+		ctrl.eqQueue.Forget(obj)
+		klog.V(5).Infof("Successfully synced %s", key)
+		return nil
+	}(keyObj)
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+	return true
+}
+
+// syncHandler syncs elastic quota and convert status.used
+func (ctrl *ElasticQuotaController) syncHandler(key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+	// Get the elastic quota resource with this namespace/name
+	eq, err := ctrl.eqLister.ElasticQuotas(namespace).Get(name)
+	if apierrs.IsNotFound(err) {
+		klog.V(5).Infof("Elastic quota has been deleted %v", key)
+		return nil
+	}
+	if err != nil {
+		klog.V(3).Infof("Unable to retrieve elastic quota %v from store: %v", key, err)
+		return err
+	}
+
+	klog.V(5).Infof("Try to process elastic quota: %s", key)
+	used, err := ctrl.computeElasticQuotaUsed(namespace, eq)
+	if err != nil {
+		return err
+	}
+	// create a usage object that is based on the elastic quota version that will handle updates
+	// by default, we set used to the current status
+	usage := eq.DeepCopy()
+	usage.Status = schedv1alpha1.ElasticQuotaStatus{
+		Used: used,
+	}
+	// Ignore this loop if the usage value has not changed
+	if apiequality.Semantic.DeepEqual(usage.Status, eq.Status) {
+		return nil
+	}
+	if _, err = ctrl.schedClient.SchedulingV1alpha1().ElasticQuotas(namespace).
+		Update(context.TODO(), usage, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	ctrl.recorder.Event(eq, v1.EventTypeNormal, "Synced", fmt.Sprintf("Elastic Quota %s synced successfully", key))
+	return nil
+}
+
+func (ctrl *ElasticQuotaController) computeElasticQuotaUsed(namespace string, eq *schedv1alpha1.ElasticQuota) (v1.ResourceList, error) {
+	used := newZeroUsed(eq)
```

Comment:
I initialize a list of zero-valued resources here to complete the status described above